packages/block-editor/src/components/use-paste-styles/index.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { getBlockType, parse } from '@wordpress/blocks';
+import { useDispatch, useRegistry } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+/**
+ * Determine if the copied text looks like serialized blocks or not.
+ *
+ * @param {string} text The copied text.
+ * @return {boolean} True if the text looks like serialized blocks, false otherwise.
+ */
+function looksLikeBlocks( text ) {
+	try {
+		const blocks = parse( text, {
+			__unstableSkipMigrationLogs: true,
+			__unstableSkipAutop: true,
+		} );
+		if ( blocks.length === 1 && blocks[ 0 ].name === 'core/freeform' ) {
+			// It's likely that the text is just plain text and not serialized blocks.
+			return false;
+		}
+		return true;
+	} catch ( err ) {
+		// Parsing error, the text is not serialized blocks.
+		// (Even though that it technically won't happen)
+		return false;
+	}
+}
+
+/**
+ * Get the "style attributes" from a given block.
+ * A "style attribute" is an attribute that doesn't have `__experimentalRole` of `content`.
+ *
+ * @param {WPBlock} block The input block.
+ * @return {Object} the filtered attributes object.
+ */
+function getStyleAttributes( block ) {
+	const blockType = getBlockType( block.name );
+	const attributes = {};
+	for ( const [ attribute, attributeType ] of Object.entries(
+		blockType.attributes
+	) ) {
+		// Mark every attribute that isn't "content" as a style attribute.
+		if ( attributeType.__experimentalRole !== 'content' ) {
+			// Apply all attributes even when they are undefined to allow overriding styles.
+			attributes[ attribute ] = Object.hasOwn(
+				block.attributes,
+				attribute
+			)
+				? block.attributes[ attribute ]
+				: attributeType.default;
+		}
+	}
+	return attributes;
+}
```

Comment:
I agree with @talldan here. Right now it can produce many unexpected results, like updating attributes that have the same name in different blocks(example `isLink` property that exists in many different core blocks, etc..)

Maybe we can start by only copying the styles from `block.supports`.

Reply:
> Maybe we can start by only copying the styles from `block.supports`.

I'll try that approach. Thanks!

Reply:
>Is there a programmatic way to obtain the same list?

I don't think so.. I guess it would be okay to start with the list for now, as new core/supports is not something that happens really often and add some comments about the need to sync.. I agree though it could be easily missed..